src/components/Main.tsx

```diff
@@ -0,0 +1,687 @@
+      />
+
+      <Stack.Screen
+        name="fioNameConfirm"
+        component={ifLoggedIn(FioNameConfirmScene)}
+        options={{
+          headerRight: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressDetails"
+        component={ifLoggedIn(FioAddressDetailsScene)}
+        options={{
+          headerTitle: () => <ParamHeaderTitle<'fioAddressDetails'> fromParams={params => params.fioAddressName} />
+        }}
+      />
+      <Stack.Screen
+        name="fioConnectToWalletsConfirm"
+        component={ifLoggedIn(FioConnectWalletConfirmScene)}
+        options={{
+          title: s.strings.title_fio_connect_to_wallet
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressSettings"
+        component={FioAddressSettingsScene}
+        options={{
+          title: s.strings.title_fio_address_settings
+        }}
+      />
+
+      <Stack.Screen
+        name="fioAddressRegisterSuccess"
+        component={ifLoggedIn(FioAddressRegisteredScene)}
+        options={{
+          headerTitle: () => <ParamHeaderTitle<'fioAddressRegisterSuccess'> fromParams={params => params.fioName} />,
+          headerLeft: () => null
+        }}
+      />
+
+      <Stack.Screen
+        name="fioDomainSettings"
+        component={FioDomainSettingsScene}
+        options={{
+          title: s.strings.title_fio_domain_settings
+        }}
+      />
+
+      <Stack.Screen name="fioRequestList" component={ifLoggedIn(FioRequestListScene)} />
+      <Stack.Screen
+        name="fioRequestApproved"
+        // @ts-expect-error
+        component={ifLoggedIn(TransactionDetailsScene)}
+        listeners={{ focus: async () => requestPermission('contacts') }}
+        options={{
+          headerTitle: () => <TransactionDetailsTitle />
+        }}
+      />
+
+      <Stack.Screen
+        name="fioSentRequestDetails"
+        component={ifLoggedIn(FioSentRequestDetailsScene)}
+        options={{
+          headerRight: () => null
+        }}
+      />
+
+      <Stack.Screen name="wcConnections" component={ifLoggedIn(WcConnectionsScene)} />
+      <Stack.Screen name="wcDisconnect" component={ifLoggedIn(WcDisconnectScene)} />
+      <Stack.Screen name="wcConnect" component={ifLoggedIn(WcConnectScene)} />
+
+      <Stack.Screen
+        name="loanDashboard"
+        component={ifLoggedIn(LoanDashboardScene)}
+        options={{
+          headerTitle: () => <EdgeLogoHeader />
+        }}
+      />
+      <Stack.Screen
+        name="loanCreate"
+        component={ifLoggedIn(LoanCreateScene)}
+        options={{
+          headerTitle: () => <EdgeLogoHeader />
+        }}
+      />
+      <Stack.Screen
+        name="loanCreateConfirmation"
+        component={ifLoggedIn(LoanCreateConfirmationScene)}
+        options={{
+          headerTitle: () => <EdgeLogoHeader />
+        }}
+      />
+      <Stack.Screen
+        name="loanDetails"
+        // @ts-expect-error
+        component={ifLoggedIn(LoanDetailsScene)}
```

Comment:
The `withLoanAccount` HOC is messed up. It took a huge amount of effort to get the `withWallet` HOC typed right, so we either need to copy that type strategy, or just leave this `// @ts-expect-error` and fix it later.

Reply:
I'm a bit lost on the type strategy difference here. I'm down for a sidebar.